internal/storage/main.go

```diff
@@ -100,9 +100,70 @@ func (c *Calculator) Calculate() {
 		return
 	}
 	for _, namespace := range namespaces.Items {
-		err := c.checkVolumesCreatePods(ctx, opLog, namespace)
-		if err != nil {
-			continue
+		opLog.Info(fmt.Sprintf("calculating storage for namespace %s", namespace.Name))
+
+		// Cleanup old calculator pods if they are still running.
+		p1, _ := labels.NewRequirement("lagoon.sh/storageCalculator", "in", []string{"true"})
+		labelRequirements := []labels.Requirement{}
+		labelRequirements = append(labelRequirements, *p1)
+		podListOption := (&client.ListOptions{}).ApplyOptions([]client.ListOption{
+			client.InNamespace(namespace.Name),
+			client.MatchingLabelsSelector{
+				Selector: labels.NewSelector().Add(labelRequirements...),
+			},
+		})
+		pods := &corev1.PodList{}
+		if err := c.Client.List(ctx, pods, podListOption); err != nil {
+			opLog.Error(err, fmt.Sprintf("error getting storage-calculator pods for namespace %s", namespace.Name))
+		}
+		for _, pod := range pods.Items {
+			c.cleanup(ctx, opLog, &pod)
+		}
+
+		// Calculate storage for volumes and databases.
+		volClaims, volErr := c.checkVolumesCreatePods(ctx, opLog, namespace)
+		if volErr != nil {
+			opLog.Error(volErr, "error calculating volumes")
+		}
+		dbClaims, dbErr := c.checkDatabasesCreatePods(ctx, opLog, namespace)
+		if dbErr != nil {
+			opLog.Error(dbErr, "error calculating databases")
+		}
+
+		// Report a "none" claim to differentiate between no storage used and no
```

Comment:
Not part of this review, just a comment generally.

I'm not really sure what the reasons for doing this `none` record were, it just came across from the older legacy storage-calculator when the rewrite happened.

All this does is add an empty `none` record in the API database every time it runs if there are no volumes. It seems like a bit of a waste, and since the storage-calculator is considerably more reliable these days than the version that used to run in core, I think whatever reasons it existed for probably aren't relevant these days?

Reply:
Nothing to do in this PR for this, just something for us to consider.